resources/frontend.rb

```diff
@@ -19,8 +19,19 @@
       variables['frontend'][new_resource.name] ||= {}
       variables['frontend'][new_resource.name]['default_backend'] ||= ''
       variables['frontend'][new_resource.name]['default_backend'] << new_resource.default_backend
-      variables['frontend'][new_resource.name]['bind'] ||= ''
-      variables['frontend'][new_resource.name]['bind'] << new_resource.bind
+      if new_resource.bind.is_a? Hash
+        variables['frontend'][new_resource.name]['bind'] = []
+        new_resource.bind.map do |addresses, ports|
+          addresses = Array(addresses) unless addresses.is_a? Array
```

Comment:
Not needed, you are using `Array(addresses)` below anyways and it will just return the array if it's already an array.